sky/backends/cloud_vm_ray_backend.py

```diff
@@ -4278,6 +4278,19 @@ def set_autostop(self,
                      down: bool = False,
                      stream_logs: bool = True) -> None:
         if idle_minutes_to_autostop is not None:
+
+            # Check if we're stopping spot
+            assert (handle.launched_resources is not None and
+                    handle.launched_resources.cloud is not None), handle
+            if handle.launched_resources.use_spot:
+                # This can be triggered by, for example:
+                #   sky launch --cloud aws --use-spot --cpus 2+ -i0 -y
+                # The cluster will be UP, the launch exited with code 1, and
+                # any stage after PRE_EXEC is not executed.
+                cloud = handle.launched_resources.cloud
+                cloud.check_features_are_supported(
+                    {clouds.CloudImplementationFeatures.STOP_SPOT_INSTANCE})
```

Comment:
Does this mean that we will only raise the error after the cluster is launched? This can be missed as people may just run `sky launch` and leave it there, it might be better to raise the error earlier before the launching actually starts.

What we did for other partially supported features is that we check the features during the failover and skip the clouds that do not support the feautures requested. Can we add this `STOP_SPOT_INSTANCE` into the requested features:
https://github.com/skypilot-org/skypilot/blob/1bb59fd25b7b2de9ccb65c52c42a356145aaddb7/sky/backends/cloud_vm_ray_backend.py#L2207-L2208

This is to fix the following case:
1. `sky launch --use-spot -i 0 echo hi` where the failover first go over AWS and then go to GCP.
2. In current implementation, we do not check the supported feature before we launch the cluster causing the error being raised after the cluster is UP, which may leave the cluster on forever, until the users check `sky status -r` themselves.
In such case, it might be better to just get rid of AWS from the failover list and directly try to run the job on GCP.

Reply:
We can do that, with a bigger refactoring in execute.py#execute(): instead of 

https://github.com/skypilot-org/skypilot/blob/1bb59fd25b7b2de9ccb65c52c42a356145aaddb7/sky/execution.py#L306-L308

We can change `requested_features` into a dict of `Resources -> set(features)`. This is because for a DAG node with `any_of`, e.g.,
```
[Task(run=<empty>)
  resources: {AWS([Spot], cpus=2+), GCP(cpus=2+)}]
```
The first resource will request STOP_SPOT while the second doesn't. 

One problem is this doesn't feel clean -- this refactoring is only for STOP_SPOT_INSTANCE while the rest of CloudImplementationFeatures are all handled cleanly. But I do not see a better way.

Wdyt?

Reply:
Hmm, when we do `sky launch -i` for a DAG, we probably need all the resources in the DAG to support the stopping.

Probably, the refactoring you mentioned before by using `cloud.check_feature_supported(resources, features)` is needed, i.e., we remove the `STOP_SPOT_INSTANCE` feature, but only decide whether a launchable resources supports `STOP` based on the field the resources have.